python/GafferUI/PresetsPlugValueWidget.py

```diff
@@ -83,4 +119,16 @@ def __menuDefinition( self ) :
 	def __applyPreset( self, unused, preset ) :
 
 		with Gaffer.UndoScope( self.getPlug().ancestor( Gaffer.ScriptNode ) ) :
+			Gaffer.Metadata.deregisterValue( self.getPlug(), "presetsPlugValueWidget:isCustom" )
 			Gaffer.NodeAlgo.applyPreset( self.getPlug(), preset )
+
+	def __applyCustomPreset( self, unused ) :
+
+		with Gaffer.UndoScope( self.getPlug().ancestor( Gaffer.ScriptNode ) ) :
+			# When we first switch to custom mode, the current value will
+			# actually be one of the registered presets. So we use this
+			# metadata value to keep track of the user's desire to be in
+			# custom mode despite of this fact. We use metadata rather than
+			# a member variable so that undo will take us back to the non-custom
+			# state automatically.
+			Gaffer.Metadata.registerValue( self.getPlug(), "presetsPlugValueWidget:isCustom", True, persistent=True )
```

Comment:
I'm pretty sure making the metadata non-persistent in FormatPlugValueWidget was deliberate. Like the comment says, it's just there to force the UI to be in custom mode for the duration of the session. If the value is actually a preset, then we want to display it via the preset name on the next reload. I think it could be argued either way, but let's match the behaviour we already have - to my knowledge nobody has complained.

Reply:
Hmm, it just seemed pretty counter-intuitive to me to have something change when you save and reopen.  I asked Ivan about it, and he said that he had just encountered this behaviour on FormatPlug and found it a bit confusing.  Lucien had no strong opinion, but when pressed to decide one way or the other, expressed a weak preference for things not changing on reload.

But it's true that if we were to do it this way, I should switch FormatPlugValueWidget as well for consistency.  I don't feel strongly about it - I can make it non-persistent if you prefer, it just seems like it's a less reliable user experience.

Reply:
But it's also misleading to reopen the file and be told that the format is "Custom" when in fact it matches an existing format (either because it was edited to match, or because a format of that size was registered since). I don't feel super strongly about which way we should go, but we should be consistent between the two. I would suggest matching the existing FormatPlug behaviour for now, rather than suck even more considerations into this massive camera overhaul...